src/semidiscretization/semidiscretization.jl

```diff
@@ -0,0 +1,17 @@
+"""
+    MetricTermsCrossProduct()
+
+Struct used for multiple dispatch on functions that compute the metric terms.
+When the argument `metric_terms` is of type `MetricTermsCrossProduct`, the 
+contravariant vectors are computed using the cross-product form.
+"""
+struct MetricTermsCrossProduct end
+
+"""
+    MetricTermsCurlInvariant()
```

Comment:
Just a small thing regarding terminology - In his 2006 paper, David Kopriva originally called this the "invariant curl form" rather than the "curl-invariant form". Some more recent papers (not by Kopriva) have indeed called it the "curl-invariant form", but I don't see why the terminology should be changed. It also may be worth adding references to the following papers:

- Kopriva, D. A. (2006). Metric identities and the discontinuous spectral element method on curvilinear meshes. Journal of Scientific Computing 26, 301-327. https://doi.org/10.1007/s10915-005-9070-8
- Vinokur, M. and Yee, H. C. (2001). Extension of efficient low dissipation high order schemes for 3-D curvilinear moving grids. In Caughey, D. A., and Hafez, M. M. (eds.), Frontiers of Computational Fluid Dynamics 2002, World Scientific, Singapore, pp. 129–164. https://doi.org/10.1142/9789812810793_0008